test/extended/machines/workers.go

```diff
@@ -122,30 +121,34 @@ var _ = g.Describe("[Feature:Machines][Disruptive] Managed cluster should", func
 		skipUnlessMachineAPIOperator(c.CoreV1().Namespaces())
 
 		g.By("validating node and machine invariants")
-		// fetch machines
-		machines, err := listMachines(dc, machineLabelSelectorWorker)
+		// fetch all machines
+		allMachines, err := listMachines(dc, "")
 		if err != nil {
 			e2e.Failf("unable to fetch worker machines: %v", err)
 		}
-		numMachineWorkers := len(machines)
-		if numMachineWorkers == 0 {
-			e2e.Failf("cluster should have worker machines")
+
+		if len(allMachines) == 0 {
+			e2e.Failf("cluster should have machines")
 		}
 
 		// fetch nodes
 		nodes, err := c.CoreV1().Nodes().List(metav1.ListOptions{
 			LabelSelector: nodeLabelSelectorWorker,
 		})
 		o.Expect(err).NotTo(o.HaveOccurred())
+		if len(nodes.Items) == 0 {
+			e2e.Failf("cluster should have nodes")
+		}
 		// map node -> machine
-		nodeToMachine, nodeMatch := mapNodeNameToMachineName(nodes.Items, machines)
+		nodeToMachine, nodeMatch := mapNodeNameToMachineName(nodes.Items, allMachines)
 		if !nodeMatch {
 			e2e.Failf("unable to map every node to machine.  nodeToMachine: %v, nodeName: %v", nodeToMachine, nodeNames(nodes.Items))
 		}
-		machineToNode, machineMatch := mapMachineNameToNodeName(machines, nodes.Items)
-		if !machineMatch {
-			e2e.Failf("unable to map every machine to node.  machineToNode: %v, machineNames: %v", machineToNode, machineNames(machines))
+		machines := make([]objx.Map, len(nodeToMachine))
+		for _, m := range nodeToMachine {
+			machines = append(machines, m)
```

Comment:
Either:
```go
machines[i] = m
```
of
```
var machines []objx.Map
for _, m := range nodeToMachine {
	machines = append(machines, m)
}

Reply:
Plus, you need to filter our any machine that manages a master node. Or, non-worker node.

Reply:
There are no non-worker node in this list. These nodes are listed using labelselector `nodeLabelSelectorWorker`

Reply:
If you know the length you are copying then using `make(<TYPE>, len)` is better as it saves reallocation in `append`.